gstreamer-sdp/src/s_d_p_attribute.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018 Sebastian Dröge <sebastian@centricular.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::mem;
+use std::ffi::CStr;
+
+use ffi;
+
+use auto::SDPResult;
+
+pub struct SDPAttribute(ffi::GstSDPAttribute);
+
+impl SDPAttribute {
+    pub fn new(key: &str, value: &str) -> Result<Self, SDPResult> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut attr = mem::uninitialized();
+            let result = from_glib(ffi::gst_sdp_attribute_set(&mut attr, key.to_glib_none().0, value.to_glib_none().0));
+            match result {
+                SDPResult::Ok => Ok(SDPAttribute(attr)),
```

Comment:
As `SDPResult` is an enum with just `OK` and `INVAL`, you can make the return values of all these functions simply `Result<_, ()>`. Less noise, and we don't have to generate the `SDPResult` type at all then (only in the -sys bindings but there we have it already)